src/main/java/pl/zalwi/ProductsController.java

```diff
@@ -0,0 +1,51 @@
+package pl.zalwi;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import pl.zalwi.data.Item;
+import pl.zalwi.data.ItemCategory;
+import pl.zalwi.data.ItemRepository;
+
+import java.math.BigDecimal;
+
+@Controller
+public class ProductsController {
+
+    ItemRepository itemRepository;
+
+    public ProductsController(ItemRepository itemRepository) {
```

Comment:
@Autowired - generalnie działa to na konstruktorze bez @Autowired (ot taki mechanizm od springa) ale zawsze staraj się to tutaj dodawać.